### ISSUE TITLE
Incorrect type narrowing for issubclass() with if statement

### ISSUE THREAD

**Issue by yaxum62** [microsoft/pyright#1175]
Example:

```python
class Base:
    x = 42


T = typing.TypeVar("T", bound=Base)


def func(cls: typing.Type[T]):
    if issubclass(cls, Base):
        cls.x  # Pyright: Cannot access member "x" for type "Never"
    else:
        cls.x  # Pyright is ok to this line.

```

**VS Code extension or command-line**
vscode extension v1.1.85

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is now addressed in 1.1.86, which I just published. It will also be included in the next release of pylance.
